streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java

```diff
@@ -202,12 +203,12 @@ protected void encodeUserEndPoint(final ByteBuffer buf,
         }
     }
 
-    private ByteBuffer encodeVersionThree() {
+    private ByteBuffer encodeVersionThree(int ver) {
```

Comment:
Ditto. Let's just add an `encodeVersionFour` with duplicated logic except the version.